### PR TITLE
Updated to support sarek3.4.2

### PR DESCRIPTION
**Pull request by kjellinjonas**

### CODE REVIEW

File: make_nf_run_script.py [hunk 1]
```diff
@@ -83,4 +89,21 @@
...
+        os.system(f"{sed_cmd} '/^wes/d;/^intervals/d' {scripts_path}/parameters.json")
+    for srch, rplc in [
+        ("_PROJECTPATH_", project_path),
+        ("_PROJECT_", project),
+        ("_GENOME_", genome)]:
+        sed_cmd = f"{sed_cmd} -e 's#{srch}#{rplc}#g'"
+
+os.system(f"{sed_cmd} {scripts_path}/params.json")
+
+
```
Comment (matrulda) [inline make_nf_run_script.py]: Nitpick: Enough with one empty line?

File: create_nf_samplesheet.sh [hunk 2]
```diff
@@ -4,18 +4,20 @@ PROJ="$1"
...
 DATAPATH="$3"
 ENVPATH="$4"
+PIPELINE="$5"
 
-if [[ $# -ne 4 ]]
+if [[ $# -ne 5 ]]
 then
   echo
-  echo "     This script requires the parameters <project name> <analysis path> <data path> <environment path>"
+  echo "     This script requires the parameters <project name> <analysis path> <data path> <environment path> pipeline"
```
Comment (matrulda) [inline create_nf_samplesheet.sh]:
```suggestion
  echo "     This script requires the parameters <project name> <analysis path> <data path> <environment path> <pipeline>"
```

Comment (matrulda) [reply create_nf_samplesheet.sh]: To stick to the same format.

File: create_nf_samplesheet.sh [hunk 3]
```diff
@@ -28,9 +30,18 @@ mkdir -p ${FQDIR}
...
 find "${DATADIR}" -name "*.fastq.gz" -exec ln -s {} "${FQDIR}/" \;
 
-python "$(find "$ENVPATH/sw/rnaseq" -name "fastq_dir_to_samplesheet.py" -print -quit)" \
-  --sanitise_name \
-  --sanitise_name_index=1 \
-  --strandedness=reverse \
-  "${FQDIR}" \
-  "${PROJDIR}/${PROJ}.SampleSheet.csv"
+
+if [[ $PIPELINE == "rnaseq" || $PIPELINE == "methylseq" ]]
```
Comment (matrulda) [inline create_nf_samplesheet.sh:34]: Is it possible to do `$PIPELINE != "sarek"` instead?

Alternatively, there could be an ELSE clause where the scripts exits with "Pipeline not supported"

File: create_sarek_samplesheet.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            sampledict[readgrp] = {"sample": sample, "R1": "", "R2": ""}
+        sampledict[readgrp][readnr] = fqpath
+
+with open(outfile, 'w') as fout:
+    fout.write("patient,sample,lane,fastq_1,fastq_2\n")
+    for readgrp in sampledict:
+        samplenm = sampledict[readgrp]["sample"]
+        R1 = sampledict[readgrp]["R1"]
+        R2 = sampledict[readgrp]["R2"]
+        entry = ",".join([samplenm, samplenm, readgrp, R1, R2])
```
Comment (matrulda) [inline create_sarek_samplesheet.py:44]: Just double-checking: Should readgrp be set as lane?

File: create_sarek_samplesheet.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        m = re.match(fcpattern, os.path.basename(os.path.dirname(fqpath)))
+        fcid = m.group(1) if m else "NA"
+
+        readgrp = f"{fcid}.{laneno}.{ssheet_idx}"  # PU, Plattform unit. Will be used as read tag in BAM-files.
+
+        if readgrp not in sampledict:
+            sampledict[readgrp] = {"sample": sample, "R1": "", "R2": ""}
+        sampledict[readgrp][readnr] = fqpath
+
+with open(outfile, 'w') as fout:
```
Comment (matrulda) [inline create_sarek_samplesheet.py:38]: You could consider using the csv library here, but there are not many benefits with it in this case imo, so I think it is fine to leave as it is.

**Review by matrulda — COMMENTED** [review]
Great work, I left some comments for you.

File: create_sarek_samplesheet.py (hunk 4 above)
Comment (kjellinjonas) [reply create_sarek_samplesheet.py:44]: yes, it's just the header that is a bit strange for the sarek samplesheets.

File: make_nf_run_script.py (hunk 1 above)
Comment (kjellinjonas) [reply make_nf_run_script.py]: removed

File: create_sarek_samplesheet.py (hunk 5 above)
Comment (kjellinjonas) [reply create_sarek_samplesheet.py:38]: I looked at it, but I did not see any real benefits either in this case

File: create_nf_samplesheet.sh [hunk 6]
```diff
@@ -28,9 +30,21 @@ mkdir -p ${FQDIR}
...
+  python "$(find "$ENVPATH/sw/rnaseq" -name "fastq_dir_to_samplesheet.py" -print -quit)" \
+    --sanitise_name \
+    --sanitise_name_index=1 \
+    --strandedness=reverse \
+    "${FQDIR}" \
+    "${PROJDIR}/${PROJ}.SampleSheet.csv"
+elif [[ $PIPELINE == "sarek" ]]
+then
+  python create_sarek_samplesheet.py \
+    ${FQDIR}" \
```
Comment (matrulda) [inline create_nf_samplesheet.sh]:
```suggestion
    "${FQDIR}" \
```

File: create_sarek_samplesheet.py
Comment (matrulda) [inline create_sarek_samplesheet.py:1]: Please run `chmod o+x` for this file so that it becomes executable.

**Review by matrulda — COMMENTED** [review]
Looks great, just found a small syntax error and a file that should be executable.

**Review by matrulda — APPROVED** [review]
:+1:

**Pull request merged**